tests/integration/challenges/utils.py

```diff
@@ -45,19 +48,46 @@ def generate_noise(noise_size) -> str:
     )
 
 
-def run_multiple_times(times):
+def run_multiple_times(times: int) -> Callable:
     """
     Decorator that runs a test function multiple times.
 
     :param times: The number of times the test function should be executed.
     """
 
-    def decorator(test_func):
+    def decorator(test_func: Callable[..., Any]) -> Callable[..., Any]:
         @wraps(test_func)
-        def wrapper(*args, **kwargs):
+        def wrapper(*args: Tuple[Any, ...], **kwargs: Dict[str, Any]) -> None:
             for _ in range(times):
                 test_func(*args, **kwargs)
 
         return wrapper
 
     return decorator
+
+
+def setup_mock_input(monkeypatch: pytest.MonkeyPatch, cycle_count: int) -> None:
+    """
+    Sets up the mock input for testing.
+
+    :param monkeypatch: pytest's monkeypatch utility for modifying builtins.
+    :param cycle_count: The number of cycles to mock.
+    """
+    input_sequence = ["y"] * (cycle_count) + ["EXIT"]
```

Comment:
Ah you can ignore my previous comment.
Neat.
2 or 3 cycles is enough to reliably finish some challenges? That seems very low.

Reply:
yeah correct, these challenges are the basic ones.